src/views/main.py

```diff
@@ -22,7 +22,17 @@ def login_required(f):
     '''Wrapper around the login_required() function of Flask, to also check the active status of the user'''
     @wraps(f)
     def decorated_function(*args, **kwargs):
-        if current_user.is_authenticated == False or check_if_user_active(current_user):
+        if current_user.is_authenticated is False or check_if_user_active(current_user):
+            return redirect(url_for('account'))
+        return f(*args, **kwargs)
+    return decorated_function
+
+
+def diaggen_role_required(f):
+    '''Wrapper around the login_required() function of Flask, to also check the active status of the user'''
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if current_user.role != "ROLE_ADMIN" and current_user.role != "ROLE_DIAGGEN":
             return redirect(url_for('account'))
         return f(*args, **kwargs)
     return decorated_function
```

Comment:
- See comment in account.html
- Maybe add role as a parameter? Like: https://github.com/mattupstate/flask-security/blob/674b18103fa8734aca71bbd084ea01e3709817ef/flask_security/decorators.py#L174
- To much for this review, but this module https://flask-security-too.readthedocs.io/en/stable/ has all these role functions build in.